zaas-client/src/main/java/org/zowe/apiml/zaasclient/service/internal/HttpsClientProvider.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.zaasclient.service.internal;
+
+import lombok.AllArgsConstructor;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.zowe.apiml.zaasclient.config.ConfigProperties;
+import org.zowe.apiml.zaasclient.exception.ZaasConfigurationErrorCodes;
+import org.zowe.apiml.zaasclient.exception.ZaasConfigurationException;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.*;
+import java.security.*;
+import java.security.cert.CertificateException;
+
+@AllArgsConstructor
+class HttpsClientProvider implements CloseableClientProvider {
+    private final RequestConfig requestConfig;
+
+    private TrustManagerFactory tmf;
+    private KeyManagerFactory kmf;
+
+    // KeyStore is initialized only in the case of PassTicket handling.
+    //     It can be null, if passticket functionality not used.
+    private final String keyStorePassword;
+    private final String keyStoreType;
+    private String keyStorePath;
+
+    public HttpsClientProvider(ConfigProperties configProperties) throws ZaasConfigurationException {
+        this.requestConfig = this.buildCustomRequestConfig();
+
+        if (configProperties.getTrustStorePath() == null) {
+            throw new ZaasConfigurationException(ZaasConfigurationErrorCodes.TRUST_STORE_NOT_PROVIDED);
+        }
+
+        initializeTrustManagerFactory(configProperties.getTrustStorePath(), configProperties.getTrustStoreType(), configProperties.getTrustStorePassword());
```

Comment:
Should you add a check for `TrustManagerFactory tmf` to be null as similar done for `KeyManagerFactory kmf` before initialising it?

Reply:
Nope, tmf is always valid and not null otherwise ZaasConfigurationException would have been thrown in constructor. 

I would do the same for kmf, but there it is used only in case of passtickets and as such could be ommited for a valid reason